rfc822/header.go

```diff
@@ -218,7 +218,7 @@ func ParseHeader(header []byte) *Header {
 		split := splitLine(line)
 
 		switch {
-		case len(bytes.TrimSpace(line)) == 0:
+		case len(bytes.Trim(line, "\r\n")) == 0:
```

Comment:
Shouldn't this be wrapped with `bytes.TrimSpace(bytes.Trim(line, "\r\n")) ?

Reply:
The header line which contains just a whitespace is allowed:

```
Subject: which has line with space
␣
 is allowed
````

`TrimSpace` would trim also space and tab symbols, which is the issue I tried to fix.

The correct header / body separator should be empty line `\r\n\r\n`.